aips/aip-x.md

```diff
@@ -0,0 +1,128 @@
+---
+aip: (this is determined by the AIP Manager, leave it empty when drafting)
+title: Batched Execution Transaction
+author: @runtian-zhou
+discussions-to (*optional): <a url pointing to the official discussion thread>
+Status: Draft
+last-call-end-date (*optional): 06/25/2024
+type: Framework
+created: 06/11/2024
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): <AIP number(s)>
+---
+
+# AIP-X - Batched Execution Transaction
+  
+
+## Summary
+
+Right now the users can interact with a Move contract via entry point function or script. Script can be a nice way of chaining multiple smart contract calls yet require users to compile scripts on their own end, which would be a problem for a light weight client setup.
```

Comment:
We should build on the motivation here.

The existing paradigm for developing on Aptos requires end developers to contemplate all the ways in which a user may interact with their application. This has many implications such that an application cannot introduce support for new asset classes without being updated and that interactions between libraries must be defined completely on-chain. For example, objects allow for ungated transfers and transfers with `TransferRef`; however, a developer cannot seamlessly introduce asset specific transfers with `TransferRef` without support from applications. Applications then need to tailor their support around these assets. FungibleAsset addressed this by introducing dispatchable function, but this does not lend itself to a general solution. Secondly, as a user, an `Entry` function is opaque and there is no way to be explicit from the user about the intent of their transaction, such that, I want to extract `5 apt` from my account to acquire `Aptomingo X`. Instead a user must review simulation of the transaction and hope that the simulation outcome is reasonably deterministic.

This AIP introduces mechanisms that push interactions on the blockchain to be a transaction generation time decision to address these concerns and make user and developer lives on Aptos better by making it easier to build applications and safely determine what operations will be performed.